### PR TITLE
Updating the ipc client based on the new client framework

### PR DESCRIPTION
**Pull request by anutosh491**

### CODE REVIEW

File: test/client_ipc.cpp [hunk 1]
```diff
@@ -6,34 +6,36 @@
...
 int main(int, char**)
 {
-    zmq::context_t context;
-    zmq::socket_t cli(context, zmq::socket_type::dealer);
-    xeus::xguid socket_id = xeus::new_xguid();
-    //cli.set(ZMQ_IDENTITY, socket_id.c_str(), socket_id.size());
-    cli.set(zmq::sockopt::linger, xeus::get_socket_linger());
-    cli.connect(xeus::get_end_point("ipc", "localhost", "shell"));
+    auto context_ptr = xeus::make_zmq_context();
+    xeus::xcontext& context = *context_ptr;
```
Comment (anutosh491) [inline test/client_ipc.cpp]: I just realized the constructor for `xcontext` is defined in the protected block is xeus. So I guess this is the way we can use a xcontext reference ?

Comment (JohanMabille) [reply test/client_ipc.cpp]: Yes, `xcontext` is an abstract class used for type erasing the right type. No need to capture the reference though, you can directly dereference the pointer when you need it.

File: test/client_ipc.cpp [hunk 2]
```diff
@@ -6,34 +6,36 @@
...
+    config.m_control_port = "control";
+    config.m_shell_port = "shell";
+    config.m_stdin_port = "stdin";
+    config.m_iopub_port = "iopub";
+    config.m_hb_port = "heartbeat";
+    config.m_signature_scheme = "none";
+    config.m_key = "";
 
-    auto auth = xeus::make_xauthentication("none", "");
+    xeus::xipc_client ipc_client(context, config);
```
Comment (JohanMabille) [inline test/client_ipc.cpp]: Here you could write `xeus::xipc_client ipc_client(*context_ptr, config)`;

Comment (anutosh491) [reply test/client_ipc.cpp]: Addressed

**Comment by JohanMabille** [comment]
Sweet!

**Pull request merged**